report/Sprint_1/Backend.tex

```diff
@@ -0,0 +1,125 @@
+        public string AppUrl { get; set; }
+        public string AppColor { get; set; }
+    }
+}
+\end{lstlisting}
+
+Each endpoint has its own controller which is used to interface with the business logic of the API.
+An example of this can be seen in~\autoref{APIAppController}.
+This snippet shows an endpoint where if the API is called on App this an HttpGet request, this function will be called and the answer returned.
+The function it self i rather simple, it calls the GetAll function and maps the result to a meaningful result that is then returned.
+
+\begin{lstlisting}[caption={API App Controller GetAll Function}, label={APIAppController}, language={CSharp}]
+/// <summary>
+/// Get All Apps in the system
+/// </summary>
+/// <returns>HttpCode 200 with a list of apps</returns>
+[HttpGet]
+public IActionResult GetAll()
+{
+    var app = _appService.GetAll();
+    var model = _mapper.Map<IList<AppModel>>(app);
+    return Ok(model);
+}
+\end{lstlisting}
+
+The actual function dealing with the logic of such a call can be seen on~\autoref{APIAppService}.
+Since this function is rather simple overall the only actual logic is accomplished by returning all apps in the datacontext, which is just a call to the database.
+
+\begin{lstlisting}[caption={API App Controller GetAll Function}, label={APIAppService}, language={CSharp}]
+/// <summary>
+/// Get All Apps
+/// </summary>
+/// <returns>List of all apps</returns>
+public IEnumerable<App> GetAll()
+{
+    return _context.Apps;
+}
+\end{lstlisting}
+
+Similar model, services and controllers can be found for each element of the list above.
+
+\subsection{Database}
+
+In~\autoref{subsec:ConclusionOfLeverage} it was concluded that PostgreSQL should be used as the datastore for this project.
+Based on previous experience the database was setup using Docker which allows the developers to run the database locally and have the exact same environment as the production servers would run.
+
+On DockerHub there are an official repository from PostgreSQL which provides the docker image to run\cite{DockerHubPostgreSQL}.
+
+With the use of entity framework in the API, the state of the database is easily maintained as migrations for the API can be added and removed as needed, meaning that the state of the database would always be functional for the API\cite{EntityFramework}.
+
+\subsection{API Documentation}
+
+Since different developes are responsible for developing different part of the solution, the documentation is a key point that enables one developer to work on the work made by others.
+Since the API is developed using the dotnet framework we can use a tool called Swagger to auto generate an interactive interface containing the API documentation\cite{SwaggerIO}.
+
+On~\autoref{SwaggerDocumentation} a screenshot of the documentation can been seen.
+With this interface the different developers where able to use the API at any given moment since the documentation would always be of the most resent version.
```

Comment:
```suggestion
With this interface, the different developers can use the API at any given moment since the documentation would always be of the most recent version.
```